pkg/userserver/user_server.go

```diff
@@ -0,0 +1,269 @@
+package userserver
+
+import (
+	"context"
+	"crypto/sha256"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"net"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/spf13/cobra"
+	"google.golang.org/grpc"
+	grpccredentials "google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/keepalive"
+	"k8s.io/klog/v2"
+	addonutils "open-cluster-management.io/addon-framework/pkg/utils"
+	"open-cluster-management.io/cluster-proxy/pkg/constant"
+	"open-cluster-management.io/cluster-proxy/pkg/utils"
+	konnectivity "sigs.k8s.io/apiserver-network-proxy/konnectivity-client/pkg/client"
+	"sigs.k8s.io/apiserver-network-proxy/pkg/util"
+
+	addonclient "open-cluster-management.io/api/client/addon/clientset/versioned"
+	addoninformers "open-cluster-management.io/api/client/addon/informers/externalversions"
+	addonlisterv1alpha1 "open-cluster-management.io/api/client/addon/listers/addon/v1alpha1"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+func NewUserServerCommand() *cobra.Command {
+	userServer := newUserServer()
+
+	cmd := &cobra.Command{
+		Use:   "user-server",
+		Short: "user-server",
+		Long:  `A http proxy server, receives http requests from users and forwards to the ANP proxy-server.`,
```

Comment:
I think it runs on the hub, would you clarify it on the description?

Reply:
Changed to `A http proxy server running on the hub cluster, ...`